apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -387,7 +387,8 @@ where
     /// Checks if we have found the correct number of Ethereum events
     /// vote extensions in [`DigestCounters`].
     fn has_proper_eth_events_num(&self, c: &DigestCounters) -> bool {
-        self.storage.last_height.0 > 0 && c.eth_ev_digest_num == 1
+        self.storage.last_height.0 == 0
+            || self.storage.last_height.0 > 0 && c.eth_ev_digest_num == 1
```

Comment:
if the height `H` is such that `H = 0` or `H > 0`, and `H` is an unsigned integer, we can just remove the first two checks altogether, and only keep the counter check.

I don't understand how this is an issue though. genesis shouldn't call process proposal, right?

Reply:
ProcessProposal is called for every block (including the genesis block). So atm when running the ledger, it gets stuck proposing the first block.

> if the height H is such that H = 0 or H > 0, and H is an unsigned integer, we can just remove the first two checks altogether, and only keep the counter check.

Oh yeah the `self.storage.last_height.0 > 0` is redundant, we still need a special check for when we're proposing the genesis block (i.e. `self.storage.last_height.0 == 0`) though, have updated